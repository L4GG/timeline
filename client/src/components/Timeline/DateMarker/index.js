// @flow
import React from 'react';
import styled from 'styled-components';

import { color, font } from '../../../styles';
import { WIDTH as AXIS_WIDTH } from '../Axis';

type Props = {|
  year: number,
  y: number,
|};

const GUTTER = 10;
const HEIGHT = 1;
const WIDTH = 40;

const TWO = 2;

const Marker = styled.div`
  background: ${color.dustyGrey};
  height: ${HEIGHT}px;
  left: 0;
  margin-left: -${(WIDTH - AXIS_WIDTH) / TWO}px;
  position: absolute;
  width: ${WIDTH}px;
`;

const Year = styled.span`
  display: block;
  font-family: ${font.family.body};
  font-size: ${font.size.small}px;
  position: absolute;
  right: ${WIDTH + GUTTER}px;
  top: 0;
  transform: translateY(-50%);
`;

const DateMarker = ({ year, y }: Props) => {
  return (
    <Marker style={{ top: y }}>
      <Year>{year}</Year>
    </Marker>
  );
};

export default DateMarker;
