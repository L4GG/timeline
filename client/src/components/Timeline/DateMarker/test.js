// @flow
/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import DateMarker from '.';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<DateMarker y={0} year={1990} />, div);
});

it('renders correctly', () => {
  const tree = renderer.create(<DateMarker y={0} year={1990} />).toJSON();

  expect(tree).toMatchSnapshot();
});
