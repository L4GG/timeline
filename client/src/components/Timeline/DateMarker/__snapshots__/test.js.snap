// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bwzfXH cFlATe"
  style={
    Object {
      "top": 0,
    }
  }
>
  <span
    className="sc-htpNat dkYnPZ"
  >
    1990
  </span>
</div>
`;
